Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -230,9 +230,12 @@ class BrowserViewController: UIViewController {
         guard let rewards = rewards, let observer = rewardsObserver else { return }
         rewards.ledger.add(observer)
         observer.fetchedPanelPublisher = { [weak self] publisher, tabId in
-            guard let self = self, self.isViewLoaded, let tab = self.tabManager.selectedTab, tab.rewardsId == tabId else { return }
-            self.publisher = publisher
-            self.updateRewardsButtonState()
+            guard let strongSelf = self, strongSelf.isViewLoaded, let tab = strongSelf.tabManager.selectedTab, tab.rewardsId == tabId else {
+                self?.resetRewardsButtonVerifiedState()
```

Comment:
I don't think this solves the underlying issue, which is that when new tabs are being shown the `fetchedPanelPublisher` observer method is reporting the old tab as the fetched publisher. This is possibly a bug in brave-core side of things (`onTabRetrieved`/reporting methods in BATBraveLedger.mm)

Steps to see this: 
1. Breakpoint on line 233 (guard statement)
2. Open a tab, go to amazon.com (will break showing tabId == selected tab)
3. Open a new tab, breaks again but the tabId being shown is still amazon.com's tab, and should be the new tab